src/linux/pci_devices.rs

```diff
@@ -0,0 +1,112 @@
+use std::{
+    fs::{read_dir, read_to_string},
+    io,
+    path::{Path, PathBuf},
+};
+
+use pciid_parser::{schema::SubDeviceId, Database};
+
+use crate::extra::pop_newline;
+
+fn parse_device_hex(hex_str: &str) -> String {
+    pop_newline(hex_str).chars().skip(2).collect::<String>()
+}
+
+pub enum PciDeviceReadableValues {
+    Class,
+    Vendor,
+    Device,
+    SubVendor,
+    SubDevice,
+}
+
+impl PciDeviceReadableValues {
+    fn as_str(&self) -> &'static str {
+        match self {
+            PciDeviceReadableValues::Class => "class",
+            PciDeviceReadableValues::Vendor => "vendor",
+            PciDeviceReadableValues::Device => "device",
+            PciDeviceReadableValues::SubVendor => "subsystem_vendor",
+            PciDeviceReadableValues::SubDevice => "subsystem_device",
+        }
+    }
+}
+
+#[derive(Debug)]
+pub struct PciDevice {
+    base_path: PathBuf,
+}
+
+impl PciDevice {
+    fn new(base_path: PathBuf) -> PciDevice {
+        PciDevice { base_path }
+    }
+
+    fn _read_value(&self, readable_value: PciDeviceReadableValues) -> String {
+        let value_path = self.base_path.join(readable_value.as_str());
+
+        match read_to_string(&value_path) {
+            Ok(hex_string) => parse_device_hex(&hex_string),
+            _ => panic!("Could not find value: {:?}", value_path),
+        }
+    }
+
+    pub fn is_gpu(&self, db: &Database) -> bool {
+        let class_value = self._read_value(PciDeviceReadableValues::Class);
+        let first_pair = class_value.chars().take(2).collect::<String>();
+
+        match db.classes.get(&first_pair) {
+            Some(class) => class.name == "Display controller",
```

Comment:
How would I do that? `db.classes` appears to be a hash map of `String` to `Class` I think

Reply:
I'll have a look at the other library I linked originally as it may be better anyway since it bundles the `pci.ids`

Reply:
Ok makes sense. Any suggestions on what I could try then since the hashmap is built using strings as keys? Would you prefer I tried making a PR for that library to change it before we use it here?

Reply:
Alright cool. So then is there anything else you would like from this PR? I can have a look at making a PR for that library to use maybe `u8` for the keys if you think that would help? Also are there any changes to documentation / testing I should make here